### ISSUE TITLE
[TODO] VST support

### ISSUE THREAD

**Issue by itsmattkc** [olive-editor/olive#1342]
<a href="https://github.com/itsmattkc"><img src="https://avatars1.githubusercontent.com/u/34096995?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [itsmattkc](https://github.com/itsmattkc)**
_Saturday May 02, 2020 at 13:36 GMT_
_Originally opened as https://github.com/sobotka/olive/issues/67_

----

**General TODO Summary**

Re-implement the ability to use external audio plugins.

**Additional Information / Notes**

I'm aware this is somewhat controversial, but I believe it could be done cleanly (confining it within a node and isolating it to a separate process) and open up functionality without disservicing the rest of the program.

**Comment by Cambloid** [comment]
Zrythm (https://www.zrythm.org/en/index.html) uses something called carla:

https://kx.studio/Applications:Carla

for audio plugins

**Comment by F0rth** [comment]
> Zrythm (https://www.zrythm.org/en/index.html) uses something called carla:
> 
> https://kx.studio/Applications:Carla
> 
> for audio plugins

Carla works very well and its cross-platform support is top notch (ie running windows only vst on linux abi).
FalkTX also wrote [Ildaeil](https://github.com/DISTRHO/Ildaeil) which might be of interest.

**Comment by alex-tee** [comment]
Hi I'm the Zrythm developer. I think it would benefit us (Olive, Zrythm, OBS, etc.) all if we shared the plugin hosting implementation in a single library (Carla) instead of implementing separate support for it - that way we can share bugfixes and improvements (ping @falktx)

Carla also already supports bridging (loading plugins in a separate process)

**Comment by falkTX** [comment]
Developer documentation for Carla as API is at https://kx.studio/ns/dev-docs/CarlaBackend/modules.html btw

**Comment by itsmattkc** [comment]
Yeah I've played around with it and am pretty sold on using it when we get around to implementing this

**Comment by DCubix** [comment]
I'm interested in this. Currently in need for a compressor and EQ. I recorded my voice overs and found out there was only a volume filter 😅

**Comment by PROPHESSOR** [comment]
This is a fantastic idea, as it will allow you to perform any manipulations with sound in one shot!
If you add support for Carla, you will not need to write modules for sound processing yourself (equalizer, compressor, limiter, reverb, delay, etc.).
It is enough to use existing plugins, which will colossally increase the capabilities of this wonderful editor.
